File: ruby/verbs/redirect.rb
require_relative 'xml_verb'

module RubyBandwidthBxml
  # The Redirect verb is used to redirect the current XML execution to another URL
  class Redirect
    include XmlVerb

    def to_xml(xml)
      xml.Redirect(compact_hash({
       'redirectUrl' => redirect_url,
       'redirectMethod' => redirect_method,
       'tag' => tag
      }))
    end
  end
end

File: csharp/BandwidthBXML/src/BandwidthBXML/verbs/Redirect.cs
﻿using System.ComponentModel;
using System.Xml.Serialization;

namespace BandwidthBXML
{
  /// <summary>
  ///   The Redirect verb is used to redirect the current XML execution to another URL.
  /// </summary>
  /// <seealso href="http://ap.bandwidth.com/docs/xml/redirect/" />
  public class Redirect : IVerb
  {
    /// <summary>
    ///   Relative or absolute URL to send event and request new BaML
    /// </summary>
    [XmlAttribute("redirectUrl")]
    public string RedirectUrl { get; set; }

    /// <summary>
    /// HTTP method for the redirect url callback
    /// </summary>
    [XmlAttribute("redirectMethod")]
    public string RedirectMethod { get; set; }

    /// <summary>
    /// Optional custom string to include in callbacks
    /// </summary>
    [XmlAttribute("tag")]
    public string Tag { get; set; }
  }
}
